### ISSUE TITLE
nsq_to_http: 'http-client-request-timeout' flag has no effect

### ISSUE THREAD

**Issue by kamyanskiy** [nsqio/nsq#998]
I'm trying to use **nsq_to_http** to make POST request when message appears in NSQ to my Flask API. I've tried to increase "http-client-request-timeout" duration that defaults 20s up to 60 sec.
But seems it doesn't take effect.

1. I added time.sleep(30) to POST request handler in my API, to check that **nsq_to_http** client waits more than 20s when it makes POST with message to my API.

2. **nsq_to_http** starts with command line:
```
- /nsq_to_http --lookupd-http-address=nsqlookupd:4161 -channel ${CHANNEL:-"channel"} -topic ${TOPIC:-"topic"} -post ${API_URL:-"http://172.16.7.87:5000/ingoing"} -content-type application/json --consumer-opt=max_attempts,1000 --http-client-connect-timeout=60s --http-client-connect-timeout=30s
```
### EXPECTED RESULT
nsq_to_http client timeout was increased upto 60 sec , so it should waits successfuly when api is not responded for 30 sec

### ACTUAL RESULT

nsq_to_http HTTPPost client drops connection after 20 sec delay.
In logs I see: 
```
nsq_to_http1_1  | 2018/02/21 15:35:27 INF    1 [topic/channel] (31a6b893668b:4150) connecting to nsqd
...
nsq_to_http1_1  | 2018/02/21 15:35:47 ERR    1 [topic/channel] Handler returned error (Post http://172.16.7.87:5000/ingoing: net/http: request canceled (Client.Timeout exceeded while awaiting headers)) for msg 09506159c6b4c000
nsq_to_http1_1  | 2018/02/21 15:35:47 WRN    1 [topic/channel] backing off for 2s (backoff level 1), setting all to RDY 0
nsq_to_http1_1  | 2018/02/21 15:35:49 WRN    1 [topic/channel] (31a6b893668b:4150) backoff timeout expired, sending RDY 1
```

**Comment by ploxiln** [comment]
I _think_ this is because the flag value is used in an `init()` function which runs _before_ the flags are parsed:

```go
func init() {
	httpclient = &http.Client{Transport: http_api.NewDeadlineTransport(*httpConnectTimeout, *httpRequestTimeout), Timeout: *httpRequestTimeout}
	userAgent = fmt.Sprintf("nsq_to_http v%s", version.Binary)
}
```

**Comment by mreiferson** [comment]
ahhh, good catch!

**Comment by kamyanskiy** [comment]
@ploxiln thank you for answer, happy to bring something to NSQ. I'm newbie in Go language and, am I correct understand - that in Go init() starts after variable initialization but before main() function. Config flags are parsed in main() function [here](https://github.com/nsqio/nsq/blob/master/apps/nsq_to_http/nsq_to_http.go#L172) ?

**Comment by ploxiln** [comment]
yes, you understand correctly

**Comment by mreiferson** [comment]
fixed in #1000
